subsys/net/ip/net_shell.c

```diff
@@ -1301,22 +1301,20 @@ static void dns_result_cb(enum dns_resolve_status status,
 {
 	struct net_shell_user_data *data = user_data;
 	const struct shell *shell = data->shell;
-	bool *first = data->user_data;
+	struct k_sem *sem = data->user_data;
+printk("in cb, status=%d, sem=%p, isr=%d\n", status, sem, k_is_in_isr());
```

Comment:
> minor: missing tab

Sorry if I didn't make it clear, but this PR is a throw-away example showing deadlock in the shell. The outcome of it can be:

1. You show me what I'm doing wrong.
2. You reinstate the functionality of shell_command_enter().
3. We indeed throw this out and go back to https://github.com/zephyrproject-rtos/zephyr/pull/13191